pkg/apis/pipeline/internal/checksum/checksum.go

```diff
@@ -0,0 +1,57 @@
+package checksum
+
+import (
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// SignatureAnnotation is the key of signature in annotation map
+	SignatureAnnotation = "tekton.dev/signature"
+)
+
+// PrepareObjectMeta will remove annotations not configured from user side -- "kubectl-client-side-apply" and "kubectl.kubernetes.io/last-applied-configuration"
+// (added when an object is created with `kubectl apply`) to avoid verification failure and extract the signature.
+// Returns a copy of the input object metadata with the annotations removed and the object's signature,
+// if it is present in the metadata.
+func PrepareObjectMeta(in metav1.Object) metav1.ObjectMeta {
+	outMeta := metav1.ObjectMeta{}
+
+	// exclude the fields populated by system.
+	outMeta.Name = in.GetName()
+	outMeta.GenerateName = in.GetGenerateName()
+	outMeta.Namespace = in.GetNamespace()
+
+	if in.GetLabels() != nil {
+		outMeta.Labels = make(map[string]string)
+		for k, v := range in.GetLabels() {
+			outMeta.Labels[k] = v
+		}
+	}
+
+	outMeta.Annotations = make(map[string]string)
+	for k, v := range in.GetAnnotations() {
+		outMeta.Annotations[k] = v
+	}
```

Comment:
I was thinking about this
```suggestion
	if in.GetLabels() != nil {
		outMeta.Labels = make(map[string]string)
		for k, v := range in.GetLabels() {
			outMeta.Labels[k] = v
		}
	}
        if in.GetAnnotations() != nil {
	    outMeta.Annotations = make(map[string]string)
	    for k, v := range in.GetAnnotations() {
		outMeta.Annotations[k] = v
	    }
        }
        delete(outMeta.Annotations, "kubectl-client-side-apply")
        other deletes ...
```